src/Perlang.Interpreter/Typing/TypesResolvedValidator.cs

```diff
@@ -355,28 +378,19 @@ public override VoidObject VisitVarStmt(Stmt.Var stmt)
                     }
                 }
             }
-            else if (stmt.Initializer == null)
+            else if (!stmt.TypeReference.ExplicitTypeSpecified && stmt.Initializer == null)
```

Comment:
(This fixes a bug. With other changes in this PR, we would sometimes get into this branch when an explicit type was set _but the type did not exist_. In other words, `ExplicitTypeSpecified` would be `true` but `IsResolved` would be false. The fix here is the right thing - we should only produce errors about type inference when type inference is attempted, of course... :see_no_evil:)